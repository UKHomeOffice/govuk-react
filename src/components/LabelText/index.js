// https://github.com/alphagov/govuk-frontend/tree/master/src/components/

import glamorous from "glamorous";
import { BLACK } from "govuk-colours";
import { MEDIA_QUERIES } from "../../constants/index";

const LabelText = glamorous.span(
  {
    fontFamily: '"nta", Arial, sans-serif',
    WebkitFontSmoothing: "antialiased",
    MozOsxFontSmoothing: "grayscale",
    display: "block",
    clear: "none",
    fontWeight: 400,
    fontSize: "16px",
    lineHeight: "1.25",
    color: `${BLACK}`,
    paddingBottom: "2px",
    [MEDIA_QUERIES.LARGESCREEN]: {
      fontSize: "19px",
      lineHeight: "1.31579"
    }
  },
  ({ error }) => ({
    fontWeight: error ? 700 : LabelText.fontWeight
  })
);

export default LabelText;
