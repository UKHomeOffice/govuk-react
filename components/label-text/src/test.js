import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import { createSerializer } from 'jest-emotion';
import * as emotion from 'emotion';

import LabelText from './';

expect.addSnapshotSerializer(createSerializer(emotion));

describe('LabelText', () => {
  let props;
  const example = 'example';
  const wrapper = <LabelText error={example}>{example}</LabelText>;
  const noError = <LabelText>{example}</LabelText>;

  beforeEach(() => {
    props = {
      children: example,
    };
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(wrapper, div);
  });

  it('passes `props.children` to the rendered `wrapper` as `children`', () => {
    expect(wrapper.props.children).toBe(props.children);
  });

  it('matches wrapper snapshot', () => {
    expect(mount(wrapper)).toMatchSnapshot('wrapper mount');
  });

  it('matches noError snapshot', () => {
    expect(mount(noError)).toMatchSnapshot('no error mount');
  });
});
