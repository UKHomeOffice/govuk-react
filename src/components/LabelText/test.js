import React from "react";
import ReactDOM from "react-dom";
import { mount } from "enzyme";

import LabelText from "./";

describe("LabelText", () => {
  const example = "example";
  const wrapper = <LabelText>{example}</LabelText>;

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(wrapper, div);
  });

  it("matches snapshot", () => {
    expect(mount(wrapper)).toMatchSnapshot(`enzyme.mount`);
  });
});
