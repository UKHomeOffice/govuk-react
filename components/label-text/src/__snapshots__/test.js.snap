// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelText matches noError snapshot: no error mount 1`] = `
.emotion-0 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  display: block;
  clear: none;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  color: #0b0c0c;
  padding-bottom: 2px;
}

@media only screen and (min-width:641px) {
  .emotion-0 {
    font-size: 19px;
    line-height: 1.3;
  }
}

<Styled(LabelText)>
  <LabelText
    className="emotion-1"
  >
    <Styled(span)
      className="emotion-1"
    >
      <span
        className="emotion-0"
      >
        example
      </span>
    </Styled(span)>
  </LabelText>
</Styled(LabelText)>
`;

exports[`LabelText matches wrapper snapshot: wrapper mount 1`] = `
.emotion-0 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  display: block;
  clear: none;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  color: #0b0c0c;
  padding-bottom: 2px;
  font-weight: 700;
}

@media only screen and (min-width:641px) {
  .emotion-0 {
    font-size: 19px;
    line-height: 1.3;
  }
}

<Styled(LabelText)
  error="example"
>
  <LabelText
    className="emotion-1"
    error="example"
  >
    <Styled(span)
      className="emotion-1"
      error="example"
    >
      <span
        className="emotion-0"
      >
        example
      </span>
    </Styled(span)>
  </LabelText>
</Styled(LabelText)>
`;
